### ISSUE TITLE
`npm run ios` error: Failed to start simulator: Process exited with non-zero code: 22

### ISSUE THREAD

**Issue by andrebr** [react-community/create-react-native-app#169]
## Description

Can't run my app on iOS simulator. It starts, but keep frozen on the default iOS screen, instead of keep loading and then show my app.

## Expected Behavior

Expected to see my app on iOS Simulator once I ran `npm run ios`.

## Observed Behavior

Ran `npm run ios` and got an error: `Failed to start simulator: Error: Process exited with non-zero code: 22`

```
$ npm run ios

> FlatApp@7.0.0 ios /app/demo
> react-native-scripts ios

Starting packager...
Starting simulator...
Failed to start simulator:

Error: Process exited with non-zero code: 22

Exiting...
```

Also tried with `yarn` and got the same error:

```
$ yarn run ios

yarn run v0.23.2
$ react-native-scripts ios 
Starting packager...
Starting simulator...
Failed to start simulator:

Error: Process exited with non-zero code: 22

Exiting...
✨  Done in 20.68s.
```

Also found this log `/Users/user1/Library/Logs/CoreSimulator/999FEB61-817B-4606-8C69-F0F7C6D05BF0/system.log`:

```
Apr 13 00:08:52 aper CoreSimulatorBridge[2540]: Error in registering callback port Error
Domain=NSPOSIXErrorDomain Code=22 "Invalid argument" UserInfo
{NSLocalizedDescription=Unable to register pasteboard callback port.,
NSLocalizedFailureReason=Port is invalid.}
```

I'm not running any kind of firewall, in case you're curious about `Port is invalid`.

Also tried the procedure `Simulator > Reset Content and Settings` as described on https://github.com/react-community/create-react-native-app/issues/62 and https://github.com/react-community/create-react-native-app/issues/67, but didn't work.

## Environment

This is my environment. Please, let me know if you would like to check something else.

* `npm ls react-native-scripts`: `react-native-scripts@0.0.2`
* `npm ls react-native`: `react-native@0.42.3`
* `npm ls expo`: `expo@15.1.3`
* `node -v`: `v7.9.0`
* `npm -v`: `4.2.0`, also tried with `4.4.4`
* `yarn --version`: `0.23.2`
* `watchman version`: `{"version": "4.7.0"}`

1. Operating system: `macOS 10.12.4`
2. Phone/emulator/simulator & version: `iOS Simulator 10.0`
3. XCode: `8.2.1`, also tried with `8.3.1`

## Reproducible Demo

```
user1:demo user1$ npm install -g create-react-native-app
/usr/local/bin/create-react-native-app -> /usr/local/lib/node_modules/create-react-native-app/build/index.js
/usr/local/lib
└── create-react-native-app@0.0.6 

user1:demo user1$ cd ..
user1:app user1$ create-react-native-app my-app
Creating a new React Native app in /app/my-app.

Installing packages. This might take a couple minutes.
Installing react-native-scripts...

yarn add v0.23.2
info No lockfile found.
[1/4] 🔍  Resolving packages...
warning react-native-scripts > xdl > node-uuid@1.4.8: Use uuid module instead
warning react-native-scripts > xdl > auth0-js > xtend > object-keys@0.4.0: 
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
success Saved lockfile.
success Saved 376 new dependencies.
...
✨  Done in 44.05s.
Installing dependencies using yarnpkg...

yarn install v0.23.2
[1/4] 🔍  Resolving packages...
warning react-native > xcode > node-uuid@1.4.7: Use uuid module instead
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
success Saved lockfile.
✨  Done in 63.10s.


Success! Created my-app at /app/my-app
...
user1:app user1$ cd my-app/
user1:my-app user1$ yarn run ios
yarn run v0.23.2
$ react-native-scripts ios 
Starting packager...
Starting simulator...
Failed to start simulator:

Error: Process exited with non-zero code: 22

Exiting...
✨  Done in 26.04s.
user1:my-app user1$
```

**Comment by dikaiosune** [comment]
Thanks for the very detailed report!

When you say that the simulator is frozen on the default iOS screen, does that mean that you're unable to interact with the simulator at all? Do you see the Expo app installed in the simulator's home screen?

**Comment by andrebr** [comment]
Thanks for the quick reply @dikaiosune!

Sorry, I wouldn't mean frozen, I would mean it's not loading my app.

When the simulator starts, there is a grid with 8 icons (calendar, photos, maps, wallet, reminders, news, health and settings) from iOS.

For the regular flow, I would expect to see this grid and after a few seconds my app would be shown. Instead, the simulator keeps showing me that iOS grid with those 8 icons. And they're fully functional, I can click on the Settings icon and see simulator options, for example. But there is no Expo app, just those 8 icons mentioned before.

**Comment by dikaiosune** [comment]
Interesting. This normally happens due to some bad state in the simulator. Resetting content and settings is the most common fix, but I think restarting has also helped people in the past. Have you restarted this computer after resetting the content and settings?

**Comment by andrebr** [comment]
Yeah, I've restarted the computer and also reinstalled `node` using `brew` and downgraded from XCode 8.3.1 to 8.2.1, which was working for a co-worker, but none of them fixed my environment.

Also, if I create a brand new React Native project using `react-native init demo` (from `npm install -g react-native-cli`) then `react-native run-ios`, the simulator works pretty fine and I can see and interact with my app.

**Comment by andrebr** [comment]
Fixed it by updating `Command Line Tools (macOS Sierra version 10.12) for Xcode` to version 8.3, which Apple just released.

**Comment by alnorris** [comment]
This is will happening for me, even if I've updated command line tools

**Comment by kalimsayyad** [comment]
Found that "Command Line Tools" was not selected in the Xcode settings. After selecting, it worked for me. Open your xcode -> Preferences -> Locations - > Command Line tools

Select xcode version and save

**Comment by somsin** [comment]
Failed to start simulator:

Error: Process exited with non-zero code: 194

**Comment by nimish-gupta** [comment]
Same error exited with non-zero code: 194

**Comment by nbkhope** [comment]
How can we solve this problem without having to update XCode to 8.3? I'm using 8.2.

**Comment by deot** [comment]
```
rm -rf  ~/.expo/ios-simulator-app-cache/*
```
I try it to solve this problem;

Source: https://forums.expo.io/t/error-installing-expo-on-ios-simulator/741

**Comment by jackkinsella** [comment]
I ran into this problem after Apple auto-updated my system. Even though I had xCode etc. working previously, the simulator stopped working with React Native because I had not agreed to Apple's latest terms and conditions. The fix in my case was simply to open xCode and hit "agree".

**Comment by marktorrance** [comment]
@deot comment (rm -f) worked for me without rebooting or other undue measures.

**Comment by Aydana1** [comment]
Just make sure you are in a right directory.

**Comment by abigger87** [comment]
If you are using Mojave, it is important to install watchman. Once you do, it will show you that Xcode is not sufficient for command line. You have to additionally install another component as shown below:

Install the Command Line Tools:
  xcode-select --install

**Comment by raztud** [comment]
xcode-select --install
xcode-select: error: command line tools are already installed, use "Software Update" to install updates

but i have the same problem.

**Comment by Luckyfella73** [comment]
Looks like there are quite a few reasons letting the iOS simulator crash ... I just solved my specific case and wanted to post what helped me - in case somebody else has the same issue.

My Error message was ( after running ```$ npm run ios```)
```
Starting packager...
Starting simulator...
Failed to start simulator:

Error: Process exited with non-zero code: 2
```

I solved the error by installing an older IOS (simulator) version < iOS 11 after I found a post on  the EXPO forums (from 2017) - stating that iOS 11 is not supported.
[https://forums.expo.io/t/cant-install-expo-on-ios-simulator/2299](https://forums.expo.io/t/cant-install-expo-on-ios-simulator/2299)

So I went back to xCode / preferences / Components tab and downloaded latest iOS 10 simulator version  - which was 10.3.1 in my case. Then choose that version at the Simulator app: Hardware / device / iOS 10.3 / phone version of your choice.

Hope that information is helpfull for somebody else.
